File: Data Structures & Algorithms/first-missing-positive/submission-0.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        
        sort(nums.begin(),nums.end());

        unordered_set<int>mp(nums.begin(),nums.end());
int i=0;
        for(i=1;i<=mp.size();i++){
            if(mp.find(i)==mp.end())
            return i;
        }

return i;




    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        int size=nums.length;
        if(size==0) return false;

        Arrays.sort(nums);
       
       int pre=nums[0];
        for(int i=1;i<size;i++){
            if(nums[i]==pre)
            return true;

            pre=nums[i];
        }
return false;
    }
}